### PR TITLE
Fix PMD warning and don't init to defaults

### PR DESCRIPTION
**Pull request by arturobernalg**
* Remove Redundant Field Initializer
* Use Files.newInputStream(Paths.get(fileName))
* Remove Instantiating Objects In Loops

### CODE REVIEW

File: src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java [hunk 1]
```diff
@@ -587,10 +587,10 @@ static void freeDirectBuffer(final ByteBuffer buffer) {
             .clean(); */
             final String SUN_CLASS = "sun.nio.ch.DirectBuffer";
             final Class<?>[] interfaces = buffer.getClass().getInterfaces();
+            final Object[] NO_PARAM = new Object[0];
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java:590]: You can use the same name we have in Commons Lang: EMPTY_OBJECT_ARRAY.

File: src/main/java/org/apache/commons/crypto/NativeCodeLoader.java [hunk 2]
```diff
@@ -171,7 +171,7 @@ private static File extractLibraryFile(final String libFolderForCurrentOS, final
             // Check whether the contents are properly copied from the resource
             // folder
             try (InputStream nativeInputStream = NativeCodeLoader.class.getResourceAsStream(nativeLibraryFilePath)) {
-                try (InputStream extractedLibIn = new FileInputStream(extractedLibFile)) {
+                try (InputStream extractedLibIn = Files.newInputStream(Paths.get(extractedLibFile.getPath()))) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/crypto/NativeCodeLoader.java:174]: See line 148 above, already a path I think.

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java]: Ok, changed

File: src/main/java/org/apache/commons/crypto/NativeCodeLoader.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: well seen. changed

Comment (ecki) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: Do you habe tested this for speed? I saw (and I don’t know why) performance regressions with the stream of Files.newInputStream compared to FileInputStream. Maybe that should be a separate change to introduce Path/NIO?

Comment (arturobernalg) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: > Do you habe tested this for speed? I saw (and I don’t know why) performance regressions with the stream of Files.newInputStream compared to FileInputStream. Maybe that should be a separate change to introduce Path/NIO?

Hi,
There is High Priority warning that recomend avoid FileStream. Cause garbage collection pauses

Comment (garydgregory) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: What warning? Issued by whom?

Comment (arturobernalg) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: https://pmd.github.io/latest/pmd_rules_java_performance.html#avoidfilestream
https://bugs.openjdk.java.net/browse/JDK-8080225

Comment (garydgregory) [reply src/main/java/org/apache/commons/crypto/NativeCodeLoader.java]: @arturobernalg 
Thanks for the pointers.

**Pull request merged**